tests/test_exceptions.py

```diff
@@ -104,6 +112,89 @@ def test_valid_TraversedPartialPath_nibbles(valid_nibbles):
         ((0, -1), ValueError),
     ),
 )
-def test_invalid_TraversedPartialPath_nibbles(invalid_nibbles, exception):
+def test_invalid_TraversedPartialPath_traversed_nibbles(invalid_nibbles, exception):
     with pytest.raises(exception):
-        TraversedPartialPath(invalid_nibbles, annotate_node(b''))
+        TraversedPartialPath(invalid_nibbles, annotate_node(b''), (1,))
+
+
+@pytest.mark.parametrize(
+    'valid_nibbles',
+    (
+        (0, 0, 0),
+        (0xf, ) * 128,  # no length limit on the nibbles
+    ),
+)
+@pytest.mark.parametrize('key_encoding', (compute_extension_key, compute_leaf_key))
+def test_valid_TraversedPartialPath_untraversed_nibbles(valid_nibbles, key_encoding):
+    # This exception means that the actual node key should have more than the untraversed amount
+    # So we simulate some longer key for the given node
+    longer_key = valid_nibbles + (0,)
+    node = annotate_node([key_encoding(longer_key), b'random-value'])
+    exception = TraversedPartialPath((), node, valid_nibbles)
+    assert exception.untraversed_tail == valid_nibbles
+    assert str(Nibbles(valid_nibbles)) in repr(exception)
+
+
+@pytest.mark.parametrize('key_encoding', (compute_extension_key, compute_leaf_key))
+def test_TraversedPartialPath_keeps_node_value(key_encoding):
+    node_key = (0, 0xf, 9)
+    untraversed_tail = node_key[:1]
+    remaining_key = node_key[1:]
+    node_value = b'unicorns'
+    node = annotate_node([key_encoding(node_key), node_value])
+    tpp = TraversedPartialPath(node_key, node, untraversed_tail)
+    simulated_node = tpp.simulated_node
+    assert simulated_node.raw[1] == node_value
+    if key_encoding is compute_leaf_key:
+        assert simulated_node.sub_segments == ()
+        assert simulated_node.suffix == remaining_key
+        assert simulated_node.raw[0] == compute_leaf_key(remaining_key)
+        assert simulated_node.value == node_value
+    elif key_encoding is compute_extension_key:
+        assert simulated_node.sub_segments == (remaining_key, )
+        assert simulated_node.suffix == ()
+        assert simulated_node.raw[0] == compute_extension_key(remaining_key)
+    else:
+        raise NotImplementedError
```

Comment:
Seems a bit odd to raise a `NotImplementedError` from a test case. Is this supposed to be a placeholder for covering additional test cases in the future?

Reply:
Yeah, I guess we usually say something like `raise Exception("Unreachable")`. The line of thinking was that the test needs to implement how to handle the different ways of encoding a key. But I'll change it to stick to the more team-standard way of doing it. :+1: